semeio/workflows/misfit_preprocessor/config.py

```diff
@@ -251,51 +252,40 @@ def _inject_default_t(config):
             cs.MetaKeys.ElementValidators: (_one_of(SPEARMAN_CORRELATION, AUTO_SCALE),),
             cs.MetaKeys.Default: AUTO_SCALE,
         },
-        SPEARMAN_CORRELATION: SPEARMAN_CORRELATION_SCHEMA,
-        AUTO_SCALE: AUTO_SCALE_SCHEMA,
+        "clustering": HIERARCHICAL_SCHEMA,
+        "pca": PCA_SCHEMA,
     },
 }
 
 
-_SCALING_SCHEMA = {
+@cs.transformation_msg("Detecting method.")
+def insert_method(layer):
+    if "method" in layer:
+        if len(layer) == 1:
+            return layer
+        if len(layer) == 2 and layer["method"] in layer:
+            return layer
+        raise ValueError(f"Found method in {layer}. Did not match {layer.keys}.")
+
+    if len(layer) != 1:
+        raise ValueError(f"Only one entry is expected. Got {layer.keys()}")
+
+    layer.update({"method": list(layer.keys())[0]})
+    return layer
+
+
+_WORKFLOW_SCHEMA = {
     cs.MetaKeys.Type: cs.types.NamedDict,
-    cs.MetaKeys.Description: ("Computes primary components and scales them."),
+    cs.MetaKeys.LayerTransformation: insert_method,
     cs.MetaKeys.Content: {
-        _THRESHOLD: {
-            cs.MetaKeys.Type: cs.types.Number,
-            cs.MetaKeys.Description: (
-                "Threshold used when computing primary components of the "
-            ),
-            cs.MetaKeys.Default: 0.95,
+        "auto_scale": _CLUSTERING_SCHEMA,
```

Comment:
This configuration should be a bit different, as not all the options will be available when doing `auto_scale`, could perhaps fix this with a context validation.

Reply:
I tried to understand how context validation would work here, but did not get a clear idea. Maybe we could check  this together.

Reply:
We proposed a solution for review and discussion!